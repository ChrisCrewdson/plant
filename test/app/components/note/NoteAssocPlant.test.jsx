
const React = require('react');
const App = require('../../../../app/components/App');
const store = require('../../../../app/store');
const NoteAssocPlant = require('../../../../app/components/note/NoteAssocPlant');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const { Provider } = require('react-redux');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('NoteAssocPlant', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = () => {};
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('NoteAssocPlant should be rendered', () => {
    const plantIds = ['p-1', 'p-2'];
    const plants = {
      'p-1': {
        _id: 'p-1',
        title: 'Plant One',
        isTerminated: false,
      },
      'p-2': {
        _id: 'p-2',
        title: 'Plant Two',
        isTerminated: false,
      },
      'p-3': {
        _id: 'p-3',
        title: 'Plant Three',
      },
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <NoteAssocPlant
                dispatch={storeDispatch}
                plantIds={plantIds}
                plants={plants}
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
