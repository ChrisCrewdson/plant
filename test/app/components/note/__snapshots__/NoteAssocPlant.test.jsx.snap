// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteAssocPlant NoteAssocPlant should be rendered 1`] = `
<div
  className="react-root"
>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <div>
      Associated plants:
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "marginLeft": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <label
          className={undefined}
          htmlFor="note-assoc-plant-filter"
          onClick={undefined}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Filter
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Filter...
        </div>
        <input
          disabled={false}
          id="note-assoc-plant-filter"
          name="filter"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
        
      </div>
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 12,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Plant One
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 12,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Plant Two
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "transparent",
            "borderRadius": "50%",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#ff4081",
              "border": 10,
              "borderRadius": "50%",
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 40,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": "bottom",
              "width": 40,
            }
          }
          tabIndex={0}
          title="Show Unchecked Plants"
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "#ffffff",
                    "display": "inline-block",
                    "fill": "#ffffff",
                    "height": 40,
                    "lineHeight": "40px",
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
